lib/facter/facts/freebsd/is_virtual.rb

```diff
@@ -6,7 +6,7 @@ class IsVirtual
       FACT_NAME = 'is_virtual'
 
       def initialize
-        @virtual = Facter::Util::Facts::Posix::VirtualDetector.new
+        @virtual = Facter::Util::Facts::Posix::VirtualDetector
       end
```

Comment:
I don't see any benefit in keeping the initializer just to store a class in an instance variable.

I would suggest to drop the initialize and use the class directly on line 13
```ruby
fact_value = Facter::Util::Facts::Posix::VirtualDetector.platform
```